janus_server/src/task.rs

```diff
@@ -48,17 +89,24 @@ pub struct TaskParameters {
     pub(crate) aggregator_endpoints: Vec<Url>,
     /// The VDAF this task executes.
     pub(crate) vdaf: Vdaf,
+    /// The role performed by the aggregator.
+    pub(crate) role: Role,
     /// Secret verification parameter shared by the aggregators.
-    _vdaf_verify_parameter: Vec<u8>,
+    pub(crate) vdaf_verify_parameter: Vec<u8>,
     /// The maximum number of times a given batch may be collected.
-    _max_batch_lifetime: u64,
+    pub(crate) max_batch_lifetime: u64,
     /// The minimum number of reports in a batch to allow it to be collected.
-    _min_batch_size: u64,
+    pub(crate) min_batch_size: u64,
     /// The minimum batch interval for a collect request. Batch intervals must
     /// be multiples of this duration.
-    _min_batch_duration: Duration,
-    /// HPKE configuration for the collector
-    _collector_hpke_config: HpkeConfig,
+    pub(crate) min_batch_duration: Duration,
+    /// HPKE configuration for the collector.
+    pub(crate) collector_hpke_config: HpkeConfig,
+    /// Key used to authenticate messages sent to or received from the other
+    /// aggregators.
+    pub(crate) agg_auth_key: AggregatorAuthKey,
+    /// HPKE recipient used by this aggregator to decrypt client reports
+    pub(crate) hpke_recipient: HpkeRecipient,
```

Comment:
This is kind of awkward, because `TaskParameters` now contains several values that not only not useful to the client but should be secret from the client, and yet `client::Client` has a field `TaskParameters`. I think the answer might be to define multiple structs, each representing a different protocol participant's view of the task parameters, and a further one in `datastore::models` representing a row in table `tasks`. However this PR is already large and I wanted to get opinions on how to proceed before doing that.

Reply:
+1 to defining per-participant task parameters e.g. `client::TaskParameters`, `aggregator::TaskParameters`, etc.

Reply:
I added `client::ClientParameters`, but stopped short of creating a new `aggregator::TaskParameters` or `datastore::model::TaskParameters`, because for the moment they'd be identical to `task::TaskParameters`. I think we can revisit that as the current `tasks` table gets normalized into `hpke_configs` and `aggregator_auth_keys` tables.